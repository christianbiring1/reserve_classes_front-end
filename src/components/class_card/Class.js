import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import styles from './Class.module.css';

export default function Class(props) {
  const {
    id, title, image, description,
  } = props;
  return (
    <div className={styles.class}>
      <Link to={`/class/${id}`}>
        <img src={image} alt="" />
      </Link>
      <div className={styles.classInfo}>
        <h5>{title}</h5>
        <span className={styles.seperator} />
        <p>{description}</p>
      </div>
      <div className={styles.socialIcons}>
        <a className="social_icon twitter" href="https://facebook.com/">
          <span />
        </a>
        <a className="social_icon facebook" href="https://twitter.com/">
          <span />
        </a>
        <a className="social_icon instagram" href="https://instagram.com/">
          <span />
        </a>
      </div>
    </div>
  );
}

Class.propTypes = {
  id: PropTypes.number.isRequired,
  description: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};
