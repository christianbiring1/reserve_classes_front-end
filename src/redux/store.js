/* eslint-disable import/no-named-as-default */
import { configureStore } from '@reduxjs/toolkit';
import reserveSlice from './reserveSlice';
import newGroupSlice from './newGroupSlice';
import groupSlice from './groupSlice';

const store = configureStore({
  reducer: {
    reserve: reserveSlice,
    groups: groupSlice,
    newgroups: newGroupSlice,
  },
});

export default store;
