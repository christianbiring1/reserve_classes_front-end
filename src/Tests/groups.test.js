import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Class from '../components/class_card/Class';
import store from '../redux/configureStore';

describe('Reserve page tests', () => {
  it('renders correctly', () => {
    const tree = render(
      <Provider store={store}>
        <BrowserRouter>
          <Class />
        </BrowserRouter>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
