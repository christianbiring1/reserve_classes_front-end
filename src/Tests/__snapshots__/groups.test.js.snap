// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reserve page tests renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="class"
      >
        <a
          href="/class/undefined"
        >
          <img
            alt="group"
            src="https://calm-mesa-77347.herokuapp.com/undefined"
          />
        </a>
        <div
          class="classInfo"
        >
          <h5 />
          <span
            class="seperator"
          />
          <p />
        </div>
        <div
          class="socialIcons"
        >
          <a
            class="social_icon twitter"
            href="https://facebook.com/"
          >
            <span />
          </a>
          <a
            class="social_icon facebook"
            href="https://twitter.com/"
          >
            <span />
          </a>
          <a
            class="social_icon instagram"
            href="https://instagram.com/"
          >
            <span />
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="class"
    >
      <a
        href="/class/undefined"
      >
        <img
          alt="group"
          src="https://calm-mesa-77347.herokuapp.com/undefined"
        />
      </a>
      <div
        class="classInfo"
      >
        <h5 />
        <span
          class="seperator"
        />
        <p />
      </div>
      <div
        class="socialIcons"
      >
        <a
          class="social_icon twitter"
          href="https://facebook.com/"
        >
          <span />
        </a>
        <a
          class="social_icon facebook"
          href="https://twitter.com/"
        >
          <span />
        </a>
        <a
          class="social_icon instagram"
          href="https://instagram.com/"
        >
          <span />
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
